ci/which-channel.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+# This script outputs the channel where a CI workflow wants to merge into.
+#
+# Inputs:
+#     BASE_SHA    The commit SHA of the branch where the PR wants to merge into.
+#
+# GitHub Action Outputs:
+#     CHANNEL     Target channel where the PR wants to merge into.
+
+set -euo pipefail
+
+# When `BASE_SHA` is missing, we assume it is from bors merge commit,
+# so hope `HEAD~` to find the previous commit on master branch.
+base_sha=$(git rev-parse "${BASE_SHA:-HEAD~1}")
+
+# Default to nightly channel
+CHANNEL="nightly"
+
+# Get symbolic names for the base_sha.
+# We only care about `rust-1.y.z` branches
+ref=$(git name-rev --name-only --refs="origin/rust-1.*" $base_sha)
+
+# Remove the patch version `.0` and keep only `1.y`,
+# so that rustup can install the correct patched toolchain
+if [[ "$ref" =~ ^origin/rust-(.*).0$ ]]
```

Comment:
I'm a bit uncertain about how this works, perhaps you can clarify.

Locally running `git name-rev`, my output looks like `origin/rust-1.69.0~1` where there is a `~` character. But the regex here doesn't allot for that. I'm a bit confused by that.

Also, does this work if I open a PR to a beta branch, but my commit is a few commits behind the tip of the branch?

Reply:
The assumption here is that a `base_sha` is always HEAD of`rust-1.*.0` branch. It either gets the commit from `github.event.pull_request.base.sha`, or from bors `merge-commit~1`. If this assumption is correct, I think `git name-rev` will always get a ref without `~` or `^`.

Reply:
Fixed with 6a973cf754fe4f7596656f203902c81427111453.